### ISSUE TITLE
API v3 Regression, login does not accept JSON data anymore

### ISSUE THREAD

**Issue by svenseeberg** [passbolt/passbolt_api#402]
* Passbolt Version: 3.2.0-1
* Platform and Target:
-- Operating system: Debian 10
-- PHP: 7.3.27-1~deb10u1
-- Web server: Nginx 1.14.2-2+deb10u4
-- Database server: MariaDB 10.3.27-0+deb10u1

### What you did
After installing 3.2.0 today on multiple servers, API clients fail to log in (https://github.com/shubhamdipt/passbolt-python-api/issues/8).

According to the current Swagger documentation (https://api-reference.passbolt.com/#/Auth/post_auth_login_json) I expect that a request like the following should still work(?):
```
 curl --header "Content-Type: application/json" --request POST --data '{"gpg_auth": {"keyid": "BA0067E80C80C5888ACA60134D037610AA3E16FB"}}' https://passbolt.example.com/auth/login.json
 ```

### What happened
The response is:
```
{
    "header": {
        "id": "4f016a4d-481e-441b-ab66-a31e36a53f18",
        "status": "error",
        "servertime": 1622804987,
        "action": "a3c19ad2-8920-5395-86d0-8567cb34f382",
        "message": "There is no user associated with this key. No key id set.",
        "url": "\/auth\/login.json",
        "code": 200
    },
    "body": null
}
```

A user with this fingerprint definitely exists (did not change):
```
SELECT COUNT(user_id) FROM gpgkeys WHERE fingerprint="BA0067E80C80C5888ACA60134D037610AA3E16FB";
+----------------+
| COUNT(user_id) |
+----------------+
|              1 |
+----------------+
```

### What you expected to happen
Should this still work? Am I missing something? Does the API documentation need an update? Is it a bug?

**Comment by stripthis** [comment]
Hi,

Yes it seems to be a regression, we can see it's failling with v3.2 and 3.1. We are currently testing with v2.14. Which was the last version that was working for you?

In the extension and other clients we use form data during login, so that might explain why the extension and cli are not affected / we didn't catch it early. For example, this would work:

```
curl -k --request POST -F 'data[gpg_auth][keyid]=03F60E958F4CB29723ACDF761353B5B15D9B054F' https://localhost:8443/auth/login.json
```

**Comment by stripthis** [comment]
Ok we found the culprid:
In `Application.php` the `BodyParserMiddleware` should be before the authentication middleware:
```
            ->add(new BodyParserMiddleware())
            ->add(new AuthenticationMiddleware($this))
            ->add(GpgAuthHeadersMiddleware::class)
            ->add($csrf);
```

We'll include the fix and a regression test in the next release (we'll try to squeeze a hotfix today). 
Apologies for the inconvenience.

**Comment by stripthis** [comment]
Patch
https://github.com/passbolt/passbolt_api/commit/8a1499499c0f93cd85cbdbb3b15845d167496d91

**Comment by svenseeberg** [comment]
That was fast, you're awesome. I can confirm that this patch fixes the issue.

**Comment by cedricalfonsi** [comment]
Fixed with v3.2.1
